src/tools/lldump.c

```diff
@@ -410,59 +417,64 @@ void print_index(INDEX index, INT32 *offset)
 	NRcount = nkeys(index);
 
 	//print data in 2 column  rkey,pad2 & fkey  
+	// the 0th entries are a backstop for btree searching
+	// regular key entries are 1..ix_nkeys
+	// There are a total of NOENTS (340) entries.
+	// INDEX Directory is 4096 bytes in size
 	INT32 offfkey = *offset + sizeof(index->ix_rkeys)+2;
-	for (n=0; n<=NRcount; n++) {
-		if (*(INT64*)&rkeys(index,n) != 0) {
-		    check_rkey(&akey,&rkeys(index,n));
-			printf(FMT_INT32_HEX_06 ":ix_rkeys[" FMT_INT_04 "]:'%-8.8s'  ", 
-				*offset, n, (char *)&rkeys(index,n));
-		} else {
-			// hack a value to represent a zero filled key
-			printf(FMT_INT32_HEX_06 ":ix_rkeys[" FMT_INT_04 "]:'%-8.8s'  ", 
-				*offset, n, "8 x 0x00");
+	for (n=0; n<NOENTS; n++) {
+		if (n == NRcount+1) {
+			if (n < NOENTS - 1) {
+				printf("\ndeleted/unused entries\n");
+			}
+			if (autoskip) break;
 		}
+		check_rkey(&akey,&rkeys(index,n));
+		printf(FMT_INT32_HEX_06 ":ix_rkeys[" FMT_INT_04 "]:'%-8.8s'  ", 
+				*offset, n, akey.rkey);
 		*offset += sizeof(rkeys(index,n));
 		printf(FMT_INT32_HEX_06 ":ix_fkeys[" FMT_INT_04 "]:" 
 			FMT_INT32_HEX "(%s)\n", 
 			offfkey, n, fkeys(index,n), fkey2path(fkeys(index,n)));
 		offfkey += sizeof(fkeys(index,n));
 	}
 	// process unused entries
-	printf("\n");
-	printf(FMT_INT32_HEX "-" FMT_INT32_HEX 
-		":ix_rkeys[" FMT_INT_04 "-" FMT_INT_04 
-		"] default value 0x0000000000000000\n",
-		*offset,
-		*offset+(INT32)(sizeof(rkeys(index,n))*(NOENTS-NRcount-1)- 1),
-		NRcount,(INT)NOENTS-1);
-	printf(FMT_INT32_HEX "-" FMT_INT32_HEX 
-		":ix_fkeys[" FMT_INT_04 "-" FMT_INT_04 
-		"] default value 0x00000000\n",
-		offfkey,
-		offfkey+(INT32)(sizeof(fkeys(index,0))*(NOENTS-NRcount-1) - 1),
-		NRcount,(INT)NOENTS-1);
-	// now check that they really are zero'ed
-	int found_deleted = 0;
-	for (n=NRcount+1; n<NOENTS; n++) {
-		if (*(INT64*)&rkeys(index,n) != 0 || fkeys(index,n) != 0) {
-			if (found_deleted++ == 0) { 
-				printf("deleted index rkey/fkey pairs\n");
-			}
-			if (*(INT64*)&rkeys(index,n) != 0) {
-				check_rkey(&akey,&rkeys(index,n));
-				printf(FMT_INT32_HEX_06 ":ix_rkeys[" FMT_INT_04 "]:'%-8.8s'  ", 
-					*offset, n, (char *)&rkeys(index,n));
-			} else {
-				// hack a value to represent a zero filled key
-				printf(FMT_INT32_HEX_06 ":ix_rkeys[" FMT_INT_04 "]:'%-8.8s'  ", 
-					*offset, n, "8 x 0x00");
+	if (autoskip) {
+		printf(FMT_INT32_HEX "-" FMT_INT32_HEX 
+			":ix_rkeys[" FMT_INT_04 "-" FMT_INT_04 
+			"] default value 0x0000000000000000\n",
+			*offset,
+			*offset+(INT32)(sizeof(rkeys(index,n))*(NOENTS-NRcount-1)- 1),
+			NRcount+1,(INT)NOENTS-1);
+		printf(FMT_INT32_HEX "-" FMT_INT32_HEX 
+			":ix_fkeys[" FMT_INT_04 "-" FMT_INT_04 
+			"] default value 0x00000000\n",
+			offfkey,
+			offfkey+(INT32)(sizeof(fkeys(index,0))*(NOENTS-NRcount-1) - 1),
+			NRcount+1,(INT)NOENTS-1);
+		// now check that they really are zero'ed
+		int found_deleted = 0;
```

Comment:
Shouldn't we do this validation even in non-summary mode?  I think the autoskip block should end above the validation code.